mapbox_gl_web/lib/src/mapbox_map_controller.dart

```diff
@@ -0,0 +1,547 @@
+part of mapbox_gl_web;
+
+class MapboxMapController extends MapboxGlPlatform
+    implements MapboxMapOptionsSink {
+  final List<DivElement> _mapElements = [];
+  final List<Function> _callbacks = [];
+
+  Map<String, dynamic> _creationParams;
+  MapboxMap _map;
+
+  SymbolManager symbolManager;
+  LineManager lineManager;
+  CircleManager circleManager;
+
+  bool _trackCameraPosition = false;
+  GeolocateControl _geolocateControl;
+  LatLng _myLastLocation;
+
+  String _navigationControlPosition;
+  NavigationControl _navigationControl;
+
+  @override
+  Widget buildView(
+      Map<String, dynamic> creationParams,
+      Function onPlatformViewCreated,
+      Set<Factory<OneSequenceGestureRecognizer>> gestureRecognizers) {
+    _creationParams = creationParams;
+    _callbacks.add(onPlatformViewCreated);
+    _registerViewFactory();
+    return HtmlElementView(viewType: 'plugins.flutter.io/mapbox_gl');
+  }
+
+  void _registerViewFactory() {
+    // ignore: undefined_prefixed_name
+    ui.platformViewRegistry.registerViewFactory('plugins.flutter.io/mapbox_gl',
+        (int viewId) {
+      _callbacks[viewId](viewId);
```

Comment:
I found this can call the wrong function instance at times. To replicate:
-> run example in web
-> select 'place symbol'
-> add a symbol to the map
-> click back
->click 'place symbol'
-> add a symbol to the map

On the last step, I get an error as the callback to platform view created goes to the previous instance. This has already completed it's future and tries to do it again.

Reply:
Every time build was called, a new `onPlatformViewCreated` was added to `_callbacks`.
I fix it with use of Set instead of List.
Thanks for the review!